src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,155 @@
-    @Override
-    public void add(T value) {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        T item;
+        Node<T> next;
+        Node<T> prev;
 
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
     }
 
-    @Override
-    public void add(T value, int index) {
+    public void add(T value) {
+        linkLast(value);
+    }
 
+    public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
-    @Override
     public void addAll(List<T> list) {
-
+        for (T temp : list) {
+            add(temp);
+        }
     }
 
-    @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return node(index).item;
     }
 
-    @Override
     public void set(T value, int index) {
-
+        checkElementIndex(index);
+        node(index).item = value;
     }
 
-    @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return unlink(node(index));
     }
 
-    @Override
     public T remove(T t) {
+        if (t == null) {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
+                    return null;
+                }
+            }
+        } else {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (t.equals(x.item)) {
+                    return unlink(x);
+                }
+            }
+        }
         return null;
     }
 
-    @Override
+    private void linkLast(T element) {
+        final Node<T> l = last;
+        final Node<T> newNode = new Node<>(l, element, null);
+        last = newNode;
+        if (l == null) {
+            first = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
+    }
+
+    private void linkBefore(T element, Node<T> currentNode) {
+        final Node<T> pred = currentNode.prev;
+        final Node<T> newNode = new Node<>(pred, element, currentNode);
+        currentNode.prev = newNode;
+        if (pred == null) {
+            first = newNode;
+        } else {
+            pred.next = newNode;
+        }
+        size++;
+    }
+
+    private T unlink(Node<T> node) {
+        final T element = node.item;
+        final Node<T> next = node.next;
+        final Node<T> prev = node.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            node.prev = null;
+        }
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            node.next = null;
+        }
+        node.item = null;
```

Comment:
we can write:
node = null
and then remove node.prev = null; and node.next = null;